File: Leetcode/sbgo/Leetcode367-有效的完全平方数.go
package sbgo

func isPerfectSquare(num int) bool {
	l, r := 1, num
	for l < r {
		mid := (l + r) >> 1
		if mid*mid >= num {
			r = mid
		} else {
			l = mid + 1
		}
	}
	return l*l == num
}

File: Leetcode/PY/Leetcode367-有效的完全平方数.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        l,r = 1,num
        while l < r:
            mid = l + r >> 1
            if mid * mid >= num: r = mid    
            else: l = mid + 1
        return l * l == num